src/toolkit/metadatas.cc

```diff
@@ -0,0 +1,223 @@
+#include <math.h>
+#include <stdio.h>
+#include <sstream>
+
+#include "agent.h"
+#include "context.h"
+#include "metadatas.h"
+namespace cyclus {
+namespace toolkit {
+
+enum type_hash {
+  s,  // string
+  b,  // boolean
+  i,  // int
+  u,  // uint
+  d   // double
+};
+
+std::unordered_map<std::string, int> type_map = {
+    std::make_pair("s", s),  // string
+    std::make_pair("b", b),  // boolean
+    std::make_pair("i", i),  // int
+    std::make_pair("u", u),  // uint
+    std::make_pair("d", d)   // double
+};
+
+Metadatas::Metadatas() {}
+Metadatas::~Metadatas() {}
+
+void Metadatas::RecordMetadatas(Agent* agent) {
+  std::string tblname = "Metadatas";
+  Json::Value::Members keys = metadatas.getMemberNames();
+  Json::Value::Members::const_iterator ikey = keys.begin();
+  Json::Value::Members::const_iterator ikey_end = keys.end();
+  for (; ikey != ikey_end; ++ikey) {
+    std::string value = "";
+    std::string type = "";
+    switch (metadatas[*ikey].type()) {
+      case Json::nullValue:
+        break;
+      case Json::stringValue:
+        type = "string";
```

Comment:
I feel like this loop could be more concise.. it seems very verbose at the moment. Maybe some structure exists in boost that could be helpful?